wirecard/oxid-ee
# WDPD-130: Create Selenium Happy Flow Tests for PayPal

### This PR

* Creates an acceptance test suite for testing checkout flows of different payment methods
* Adds PayPal checkout tests for purchase/authorization actions
* Adds an alias to the Docker web container to allow accessing it via an arbitrary url

### How to Test

* Run `rake runtests_selenium` and check if the two added tests run – this will take about 2 minutes
* If you want to see the tests being run visually, follow the steps in the _Visual Feedback_ section
* Confirm that the tests succeed
* Open the shop in the frontend and check if it was not modified (e.g. the theme is not "Azure")

### Visual Feedback

In order to see the Selenium tests being run in the browser, follow these steps:

1. Install a VNC client, e.g. https://www.realvnc.com/en/connect/download/viewer/
2. Using the VNC client, connect to the Selenium container via `localhost:5900` (if you changed `OXID_HOSTNAME` or `SELENIUM_VNC_PORT` in your env file, adapt the URL accordingly)
3. If a password prompt is shown, use the password `secret`
4. Start the tests and observe them in the VNC window

### Jira Links

* WDPD-130: https://jira.parkside.at/browse/WDPD-130
